pcdsdevices/attenuator.py

```diff
@@ -291,37 +304,46 @@ def format_status_info(self, status_info):
                 filter_info = status_info[f'filter{i}']
             except KeyError:
                 break
-            status = filter_info['status']['value']
+            status = get_status_value(filter_info, 'status', 'value')
             blade_states.append(status)
-        lines = render_ascii_att(blade_states)
-        txt = 'Transmission for {} harmonic (E={:.3} keV): {:.4E}'
-        energy = status_info['energy']['value'] / 1000
-        trans = status_info['position']
-        harm = self._harmonic
-        lines.append(txt.format(harm, energy, trans))
-        return '\n'.join(lines)
 
+        states = '\n'.join(render_ascii_att(blade_states))
 
-class AttBase3rd(AttBase):
-    """
-    :func:`Attenuator` class to use the 3rd harmonic values.
+        energy = get_status_value(status_info, 'energy', 'value') * 1e3
+        energy_3rd = get_status_value(status_info, 'energy_3rd', 'value')
+        trans = get_status_value(status_info, 'position')
 
-    This is exactly the same as the normal :class:`AttBase`, but with the
-    alternative transmission PVs. It can be instantiated using the
-    `~Attenuator.use_3rd` argument in the :func:`Attenuator` factory function.
-    """
+        if energy_3rd != 'N/A':
+            energy_3rd = energy_3rd * 1e3
+            return f"""\
+{states}
+Transmission for 1st harmonic (E={energy:.3} keV): {trans:.4E}
+Transmission for 3rd harmonic (E={energy_3rd:.3} keV): {trans:.4E}
+"""
+        else:
+            return f"""\
+{states}
+Transmission for 1st harmonic (E={energy:.3} keV): {trans:.4E}
+"""
 
-    _harmonic = '3rd'
 
+class AttBaseWith3rdHarmonic(FltMvInterface, PVPositioner):
```

Comment:
I don't think it causes a problem (so it doesn't need to be fixed), but this class doesn't really function as a `PVPositioner` on its own. I'd have preferred to either subclass `AttBase` only (e.g. extend the base class) or subclass `ophyd.Device` only (e.g. just define some PVs to mix in later).

Reply:
Ohh so what I can do is basically:
```
AttBaseAndWith3rdHarmonic(AttBase):
...
```
and in the `_make_att_class()` I would need to only import use this class instead of using both 

.. i think this is what you're saying, i'm going to change that then cause i like that much better, thank you

Reply:
Changed this in b56352c - I know you said it does not need to be fixed, so let me know if I should go back, but I thought it was a good idea...

Reply:
looks :+1: to me